Backendless/Transaction/TransactionHelper.cs

```diff
@@ -0,0 +1,200 @@
+    {
+      Dictionary<String, Object> referenceToObjectId;
+
+      if( OperationTypeUtil.supportCollectionEntityDescriptionType.Contains( parentObject.GetOpResult().GetOperationType() ) )
+        referenceToObjectId = parentObject.ResolveTo( "objectId" ).MakeReference();
+      else if( OperationTypeUtil.supportListIdsResultType.Contains( parentObject.GetOpResult().GetOperationType() ) )
+        referenceToObjectId = parentObject.MakeReference();
+      else
+        throw new ArgumentException( ExceptionMessage.REF_TYPE_NOT_SUPPORT );
+
+      return referenceToObjectId;
+    }
+
+    internal static void MakeReferenceToValueFromOpResult( Dictionary<String, Object> map )
+    {
+      foreach( KeyValuePair<String, Object> kvp in map )
+      {
+        Dictionary<String, Object> entry = new Dictionary<String, Object>( map );
+
+        if( entry[ kvp.Key ] is OpResult )
+          if( OperationTypeUtil.supportIntResultType.Contains( ( (OpResult) kvp.Value ).GetOperationType() ) )
+            entry[ kvp.Key ] = ( (OpResult) kvp.Value ).MakeReference();
+          else
+            throw new ArgumentException( ExceptionMessage.OP_RESULT_FROM_THIS_OPERATION_NOT_SUPPORT_IN_THIS_PLACE );
+
+        if( entry[ kvp.Key ] is OpResultValueReference )
+        {
+          OpResultValueReference reference = (OpResultValueReference) kvp.Value;
+
+          if( CreateUpdatePropName( reference ) || CreateBulkResultIndex( reference ) || FindPropNameResultIndex( reference ) )
+            entry[ kvp.Key ] = reference.MakeReference();
+          else
+            throw new ArgumentException( ExceptionMessage.OP_RESULT_FROM_THIS_OPERATION_NOT_SUPPORT_IN_THIS_PLACE );      
+        }
+      }
+    }
+
+    public static void MakeReferenceToObjecIdtFromOpResult( List<Object> listObjectIds )
+    {
+      IEnumerator<Object> iterator = listObjectIds.GetEnumerator();
+
+      while( iterator.MoveNext() )
+      {
+        Object temp = iterator.MoveNext();
+
+        if( temp is OpResult )
+          throw new ArgumentException( ExceptionMessage.OP_RESULT_FROM_THIS_OPERATION_NOT_SUPPORT_IN_THIS_PLACE );
+
+        if( temp is OpResultValueReference )
+        {
+          OpResultValueReference reference = (OpResultValueReference) temp;
+
+          if( CreateUpdateObjectId( reference ) || CreateBulkResultIndex( reference ) || FindPropNameResultIndex( reference ) )
+            temp = reference.MakeReference();
+          else
+            throw new ArgumentException( ExceptionMessage.OP_RESULT_FROM_THIS_OPERATION_NOT_SUPPORT_IN_THIS_PLACE );
+        }
+      }
+    }
+
+    private static Boolean CreateUpdateObjectId( OpResultValueReference reference )
+    {
+      return CreateUpdatePropName( reference ) && reference.GetPropName().Equals( "objectId" );
+    }
+
+    private static Boolean CreateUpdatePropName( OpResultValueReference reference )
```

Comment:
the name of the method is not intuitive since the return value is Boolean. Method name should reflect what it  does